Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1078,6 +1118,40 @@ bool LineBuilder::tryPlacingFloatBox(const InlineItem& floatItem, LineBoxConstra
     auto& floatBox = floatItem.layoutBox();
     ASSERT(formattingState());
     auto& boxGeometry = formattingState()->boxGeometry(floatBox);
+    auto containingBlockMarginTrim = floatBox.parent().style().marginTrim();
+    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
+    auto trimMarginsForPositionedFloat = [&](LayoutPoint& floatPosition) {
+        auto marginStart = boxGeometry.marginStart();
+        auto marginEnd = boxGeometry.marginEnd();
+        if (floatingContext.isFloaingCandidateLogicallyLeftPositioned(floatBox)) {
+            if (containingBlockMarginTrim.contains(MarginTrimType::InlineStart) && floatPosition.x() - marginStart == m_rootHorizontalConstraints->logicalLeft) {
+                floatPosition.moveBy({ -marginStart, 0_lu });
+                trimMarginFromFloatGeometry(boxGeometry, MarginTrimType::InlineStart);
+            }
+            if (containingBlockMarginTrim.contains(MarginTrimType::InlineEnd) && floatPosition.x() + boxGeometry.contentBoxWidth() + marginEnd == m_rootHorizontalConstraints->logicalRight())
```

Comment:
same as with the block level margin-trim, please make sure that this is what the spec says.